cmake/Modules/ArrowExternalProject.cmake

```diff
@@ -53,11 +56,27 @@ set(ARROW_CMAKE_ARGS
   -DARROW_JEMALLOC=off
   -DARROW_WITH_BROTLI=off
   -DARROW_WITH_LZ4=off
-  -DARROW_WITH_ZLIB=off
   -DARROW_WITH_ZSTD=off
   -DFLATBUFFERS_HOME=${FLATBUFFERS_HOME}
   -DBOOST_ROOT=${BOOST_ROOT})
 
+if ("${CMAKE_RAY_LANG_PYTHON}" STREQUAL "YES")
+  # PyArrow needs following settings.
+  set(ARROW_CMAKE_ARGS ${ARROW_CMAKE_ARGS}
+    -DARROW_WITH_THRIFT=ON
+    -DARROW_PARQUET=ON
+    -DARROW_WITH_ZLIB=ON)
+else()
+  set(ARROW_CMAKE_ARGS ${ARROW_CMAKE_ARGS}
+    -DARROW_WITH_THRIFT=OFF
+    -DARROW_PARQUET=OFF
+    -DARROW_WITH_ZLIB=OFF)
+endif ()
+if (APPLE)
+  set(ARROW_CMAKE_ARGS ${ARROW_CMAKE_ARGS}
+    -DBISON_EXECUTABLE=/usr/local/opt/bison/bin/bison)
```

Comment:
do not hard code this path, use the result of `which bison`

Reply:
I think the default bison path will cause CI failure. The version is not correct. That is why I added commit of "Fix Mac building error in CI". Actually, this hard code exists in many places of ray: [example1](https://github.com/ray-project/ray/blob/master/cmake/Modules/ParquetExternalProject.cmake#L35), [example2](https://github.com/ray-project/ray/blob/master/thirdparty/scripts/build_parquet.sh#L12).

Anyway, I can try it to see whether it works or not.

Reply:
@raulchen It failed with that, so I changed back.